### ISSUE TITLE
Comments linebreaks spacing issues

### ISSUE THREAD

**Issue by RedGuy12** [ScratchAddons/ScratchAddons#1444]
- When you reply to an ST member, the first text node starts with an asterisk and the following whitespace is not trimmed.

- lf a comment contains a link, a space is ***not*** added after it, whether or not there was one.

- If a comment contains a link, a space is added before it, whether or not there was one.

- Any linebreaks before/after a link are not preserved.

**Comment by TheColaber** [comment]
For issue number 1: What's the best way to do this? it creates all this annoying whitespace...
![image](https://user-images.githubusercontent.com/72760579/106279178-06e81480-620a-11eb-897e-5159f84fb0b6.png)

But I'm not getting number 2 and 3.
![image](https://user-images.githubusercontent.com/72760579/106279356-4adb1980-620a-11eb-9dd8-4f7bcd5120ab.png)

NUmber 4 might be fixable depending on how I made my code before.

**Comment by RedGuy12** [comment]
Hmm I guess 2 and 3 are only with @mention links.

There is a space before but none after with @mentions.

**Comment by RedGuy12** [comment]
For the first, we could write a RegExp: `\/*?/s*\` That finds an optional * then any whitespace

**Comment by TheColaber** [comment]
Yes but users can add "*" to the beginning of their post so this wouldn't work.

**Comment by RedGuy12** [comment]
We would have to only use the regexp for the first trim.

It requires the * to be the first thing in the comment, before the whitespace. That is how it is when replying to the ST. If your comment starts with an asterisk, it works done after the whitespace and would not be trimmed.

Or we could do a simple `strpos(*) == 0` to figure out if it is there
________________________________

**Comment by TheColaber** [comment]
Actually, the difference between a comment node replied to an ST, or made by itself, is that the whitespace before the "*"
ST comment:
![image](https://user-images.githubusercontent.com/72760579/106283823-66e1b980-6210-11eb-858d-f8ef03c14225.png)
Madeup comment:
![image](https://user-images.githubusercontent.com/72760579/106283854-706b2180-6210-11eb-8f83-898f2ee1fc2c.png)

I can just check if the index 1 of the sting is a "*" and that will let me know it's an ST comment
